sihl/lib/core/core_jwt.mli

```diff
@@ -0,0 +1,21 @@
+type algorithm = Jwto.algorithm = HS256 | HS512 | Unknown
+
+type t
+
+type payload
+
+val empty : payload
+
+val add_claim : key:string -> value:string -> payload -> payload
+
+val encode : algorithm -> secret:string -> payload -> (string, string) result
+
+val decode : secret:string -> string -> (t, string) result
+
+val get_claim : key:string -> t -> string option
+
+val pp : Format.formatter -> t -> unit
+
+val eq : t -> t -> bool
+
+module Jwto = Jwto
```

Comment:
Why do you expose this outside of the module?

Reply:
We only expose a subset that covers most of the use cases. We also expose the underlying library so that the user can use it directly if our convenient API is not powerful enough.